net/socket/getpeername.c

```diff
@@ -121,7 +123,22 @@ int psock_getpeername(FAR struct socket *psock, FAR struct sockaddr *addr,
       return -EOPNOTSUPP;
     }
 
-  return psock->s_sockif->si_getpeername(psock, addr, addrlen);
```

Comment:
Can we zero out the whole addr before call si_getpeername/si_getsockename/si_accept/si_recvfrom?

Reply:
@xiaoxiang781216 Do you mean that if (add != NULL &&  addrlen > 0), then memset(addr, 0, addrlen) ?

Reply:
Yes, something like this, but we need check each implementation don't overwrite the unused field with the random bit again.